pandas/core/reshape/merge.py

```diff
@@ -1477,6 +1475,19 @@ def get_join_indexers(
         right_keys
     ), "left_key and right_keys must be the same length"
 
+    left_n = len(left_keys[0])
+    right_n = len(right_keys[0])
+    if left_n == 0:
+        if how in ["left", "inner", "cross"]:
```

Comment:
do we have tests for all of these empty cases? if not can you add some

Reply:
I think most of these are covered but they are scattered around. I added a parameterized test that covers the different join types and left vs right being empty.